### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by codepantha**
## Changes introduced via this PR: 👇 
**User Model** 👦 
- [x] A method that returns the 3 most recent posts for a given user.

**Post Model** 📑 
- [x] A method that updates the posts counter for a user.
- [x] A method that returns the 5 most recent comments for a given post.

**Comment Model** 💬 
- [x] A method that updates the comments counter for a post.

**Like Model** ❤️ 
- [x] A method that updates the likes counter for a post.

**ApplicationController**

- [x] Create a method called current_user that returns the first user from a database.

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -1,4 +1,8 @@
 class Like < ApplicationRecord
   belongs_to :user
```
Comment (julie-ify) [inline app/models/like.rb:2]: - [ ] **[OPTIONAL]** In your migration file the `Likes` table should reference  `author_id`,  and not the `user_id` something like this `t.references :author, foreign_key: { to_table: 'users' } ` . This was supposed to be done correctly in the previous milestone.  This `belongs_to :user` is supposed to be `belongs_to :author, class_name: 'User'`.

File: app/models/post.rb [hunk 2]
```diff
@@ -2,4 +2,14 @@ class Post < ApplicationRecord
   belongs_to :user
   has_many :likes
```
Comment (julie-ify) [inline app/models/post.rb:3]: - [ ] **[OPTIONAL]** Here too  `belongs_to :user` is supposed to be `belongs_to :author, class_name: 'User'`.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 class Comment < ApplicationRecord
   belongs_to :user
```
Comment (julie-ify) [inline app/models/comment.rb:2]: - [ ] **[OPTIONAL]** Here too  `belongs_to :user` is supposed to be `belongs_to :author, class_name: 'User'`.

**Review by julie-ify — APPROVED** [review]
## Hi @codepantherr, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/like.rb (hunk 1 above)
Comment (codepantha) [reply app/models/like.rb:2]: Thank you for providing clarity on this 🙏, @julie-ify

**Pull request merged**